### PR TITLE
[QA] CloudinaryImage refactor

### PR DESCRIPTION
**Pull request by c7hudson**
**Description:**
- [Asana Task](https://app.asana.com/0/1116481661798430/1142171941519789/f)
- Drops the `cloudinary-react` package in favour of `cloudinary-core` which is the vanilla JS SDK. The `cloudinary-react` package was a blocker for us implementing lazy loading with a `data-src` or `data-srcset` attribute on the image tag. 
- Implements the `lazysizes` package to handle lazy loading, which is applied by default to all images. 
- More explicit definition of Cloudinary resource URLs through the use of `<picture>` and `<source>` tags. We no longer rely on client hints (not widely supported by browsers) and Cloudinary's automatic detection features to determine what next-gen image format (`.jp2`/`.webp`) or what DPR value to serve. All possible image resources are defined in the markup instead of relying on Cloudinary making on the fly adjustments after checking with the client.
- Reduces the amount of `JSX` required CMS side for implementing responsive images. Previously multiple `CloudinaryImage` tags would each be wrapped in a `<Media>` component with a specific viewport query. Now all 4 viewport image sizes are passed in an array to the `width`/`height` parameters and output in the appropriate `<source>` tag making use of the `media` attribute, with breakpoints imported from EDS `Media.js`.
- Alt tags are now required to be passed to `CloudinaryImage`.
- SVG images are output using just an `` tag without the extra transformation parameters that are used for `.jpg/.png` images. The element that wraps an SVG output by `CloudinaryImage` should be used to dictate size, with CSS media queries for responsiveness. 

## CHANGES 

 Okay I changed up this PR a little bit. It should now use the automatic format setting that cloudinary offers and through testing it looks like it works pretty well, this was a good call. I had to leave in the DPR setting as the auto setting in their lib was not working as well manual. 
I also used this opportunity to add in LQIP to the images which should help with first loads and content shifting around. I added a blur-up to make the transition a little smoother between the images.
This is the gist of what we build now:
* 4 source tags, one for each media query.
* 1 img tag that has the LQIP as `srcSet` and also a fallback image for old browsers in the `src`

```html
<picture>
  <source media="(min-width: 1200px)" data-srcset="https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_1.0,f_auto,fl_progressive:semi,h_624,q_auto:eco,w_636/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg 1x, https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_2.0,f_auto,fl_progressive:semi,h_624,q_auto:eco,w_636/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg 2x, https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_3.0,f_auto,fl_progressive:semi,h_624,q_auto:eco,w_636/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg 3x"/>
  <source media="(min-width: 900px)" data-srcset="https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_1.0,f_auto,fl_progressive:semi,h_428,q_auto:eco,w_428/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg 1x, https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_2.0,f_auto,fl_progressive:semi,h_428,q_auto:eco,w_428/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg 2x, https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_3.0,f_auto,fl_progressive:semi,h_428,q_auto:eco,w_428/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg 3x"/>
  <source media="(min-width: 600px)" data-srcset="https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_1.0,f_auto,fl_progressive:semi,h_332,q_auto:eco,w_332/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg 1x, https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_2.0,f_auto,fl_progressive:semi,h_332,q_auto:eco,w_332/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg 2x, https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_3.0,f_auto,fl_progressive:semi,h_332,q_auto:eco,w_332/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg 3x"/>
  <source media="(max-width: 599px)" data-srcset="https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_1.0,f_auto,fl_progressive:semi,h_250,q_auto:eco,w_375/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg 1x, https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_2.0,f_auto,fl_progressive:semi,h_250,q_auto:eco,w_375/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg 2x, https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_3.0,f_auto,fl_progressive:semi,h_250,q_auto:eco,w_375/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg 3x"/>
  <img class="Images-module--Image--1EVuH  lazyload " src="https://res.cloudinary.com/getethos/image/upload/c_fill,f_auto,fl_progressive:semi,q_auto:eco,t_unsupported/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg" srcSet="https://res.cloudinary.com/getethos/image/upload/c_fill,f_auto,fl_progressive:semi,h_250,q_auto:eco,t_lqip,w_375/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg" alt="Family playing in bed"/>
</picture>
```

As you can see this is ~2900 characters per image, vs the ~9k from before with 12 source tags. The impact is very low from my testing on FCP. A page with many images like `/blog` comes out to about 50k more characters which is zipped to ~700 bytes. 

I went pretty soft on LQIP so they are usually between 5 -10 kb each. But we could take a more aggressive approach using either a simple default placeholder, or using greyscale versions of the images . (~1.5 - 3 kb).

image

**Referencing PR or Branch (e.g. in CMS or monorepo):**
- https://github.com/getethos/cms/pull/1907 - Still WIP, have to review two modules and then should be good to go.

**Screenshots:**
- ... to come

### CODE REVIEW

File: src/components/Images/Images.js [hunk 1]
```diff
@@ -1,64 +1,347 @@
...
-      className={['Image', className].join(' ')}
-      cloudName={CLOUDINARY_CLOUD_NAME}
-      dpr="auto"
-      responsive
-      publicId={publicIdFilename}
-      crop={crop} 
-      secure="true"
-      flags="progressive:semi"
-    />
+    <picture>
```
Comment (roblevintennis) [inline src/components/Images/Images.js]: Wow, they make us configure all this? Seems like they'd let us setup some sort of simpler metadata object and just initialize and be done 🤔

**Comment by roblevintennis** [comment]
What's the status on this @c7hudson ? Do you need some help in terms of a review or direction and is this something we're still   wanting to do?

**Comment by c7hudson** [comment]
> What's the status on this @c7hudson ? Do you need some help in terms of a review or direction and is this something we're still wanting to do?

@roblevintennis I've just got a couple things to finish up with the paired CMS PR, then this will be ready for review and implementation. 

Last it had been discussed we were still planning on going this route to achieve serving next generation image formats and lazy loading of images.

File: src/components/Images/Images.js (hunk 1 above)
Comment (c7hudson) [reply src/components/Images/Images.js]: Each call to their Image Transformation API relies on building a dynamic URL with parameters specific to the resource you're requesting. 

There wasn't an option I could see in the SDK for generating a `<picture>`  tag with `<source>` tags automatically.

Comment (c7hudson) [reply src/components/Images/Images.js]: Are you referring to setting up all the different `<source>` tags manually @roblevintennis or something to do with the options we pass when generating each URL?

File: src/components/Images/Images.js [hunk 2]
```diff
@@ -1,64 +1,347 @@
...
+
+  const dpr2Setting = {
+    dpr: '2.0'
+  }
+
+  const dpr3Setting = {
+    dpr: '3.0'
+  }
+
+  // ------------------------------
```
Comment (c7hudson) [inline src/components/Images/Images.js]: These settings should probably be rolled up into a mapping of the 3 DPR settings nested within the file format and media breakpoint settings, to write out each `<source>` markup. Then we don't have the repeated constant assignments and instead something like:
```
// for each file type (webp/jp2/jpeg) output each media breakpoint as
<source media="{[arrayOfDprVariations ]}"/>

And just have to do an exception or separate output for the last tag, `<img>`, within `<picture>`. That should DRY up this code a bunch.

Comment (roblevintennis) [reply src/components/Images/Images.js]: yeah, I'd love to see it streamlined somehow so if ^ idea does that I'm all for it 👍

File: src/components/Images/Images.js [hunk 3]
```diff
@@ -67,236 +66,99 @@ export const CloudinaryImage = ({
...
+  if (height) {
+    height.reverse()
+  }
+
+  let sourceTags = []
+  const srcSetAttribute = lazyload ? 'data-srcset' : 'srcSet'
+  const srcAttribute = lazyload ? 'data-src' : 'src'
+
+  for (var format = 0; format < fileFormats.length; format++) {
+    for (
```
Comment (roblevintennis) [inline src/components/Images/Images.js]: Not usually a fan of nested loops, but I guess we know ahead these will be finite and small correct?

Also, can we create some test coverage for all the conditional logic in here and possibly make this more testable and less procedural by breaking it out into meaningful methods/functions?

File: src/components/Images/Images.js [hunk 4]
```diff
@@ -67,236 +66,99 @@ export const CloudinaryImage = ({
...
+
+        let srcAttributeObject = {}
+        srcAttributeObject[srcAttribute] = cld.url(
+          publicIdFilename,
+          imageSettings
+        )
+
+        sourceTags.push(
+          <img
+            key={`${format}-${breakpoint}`}
```
Comment (roblevintennis) [inline src/components/Images/Images.js]: uuid

File: src/components/Images/Images.js [hunk 5]
```diff
@@ -326,22 +188,18 @@ CloudinaryImage.CROP_METHODS = {
...
 CloudinaryImage.PUBLIC_PROPS = {
-  height: PropTypes.oneOfType([
-    PropTypes.array,
-    PropTypes.bool,
-  ]),
-  width: PropTypes.oneOfType([
-    PropTypes.array,
-    PropTypes.bool,
-  ]),
+  height: PropTypes.oneOfType([PropTypes.array, PropTypes.bool]),
```
Comment (roblevintennis) [inline src/components/Images/Images.js]: Why would this be boolean?? Same with Width?

File: src/components/Images/Images.md [hunk 6]
```diff
@@ -1,5 +1,14 @@
 ```jsx
 <div style={{ width: 650 }}>
+  <CloudinaryImage
+    crop="fill"
+    alt="father and kids playing"
+    width={[250,350,450,550]}
```
Comment (roblevintennis) [inline src/components/Images/Images.md]: re question about boolean. If that IS a thing, can you   show an example here or put a comment above the jsx providing for how that works?

**Comment by roblevintennis** [comment]
Also looks like we need to merge or rebase back in the base branch and resolve conflices

**Comment by roblevintennis** [comment]
@c7hudson what's the status here? This has been a WIP for quite some time now.

Also, I think Design has mentioned to us that they'd like us to remove the radius. Probably:

```
.Image {
  border-radius: 2px; // THIS GUY
  display: block;

  // These are consequential, may need to pull out.
  width: 100%;
  height: auto;
}
```

And we could use a markdown demo as well.

File: src/components/Images/Images.js [hunk 7]
```diff
@@ -42,22 +43,20 @@ export const CloudinaryImage = ({
...
 
-  // Serve a simpler version if resource is SVG
-  if(publicId.split('.').pop() === 'svg'){
+  const imageClasses = lazyload ? 'Image lazyload' : 'Image'
+  if (className) {
+    imageClasses = [imageClasses, className].join(' ')
+  }
 
+  // Serve a simpler version if resource is SVG
+  if (publicId.split('.').pop() === 'svg') {
```
[the remark this answers is not in the document]
Comment (tkarssli) [reply src/components/Images/Images.js]: I don't know if he resolved this but I believe that `publicId.split('.').pop() === 'svg'` is a good/reliable implementation

[the remark this answers is not in the document]
Comment (roblevintennis) [reply src/components/Images/Images.js]: Ok, if you've decided that sure. I'm very keen for us to get this PR working and pushed through. I'll only block on major things at this point

File: src/components/Images/Images.js [hunk 8]
```diff
@@ -40,125 +52,105 @@ export const CloudinaryImage = ({
...
-        let imageSettings = {
-          ...baseImageSettings,
-          format: fileFormats[format],
-          dpr: '1.0',
-        }
+        return cld.url(filePath(), imageSettings) + ` ${indx + 1}x`
+      })
+
+      let minMax = breakpoint < mediaBreakpoints.length - 1 ? `min` : `max`
+      sourceTags.push(<source
```
Comment (roblevintennis) [inline src/components/Images/Images.js]: You can use `const` if you're only pushing and not literally overwriting this

File: src/components/Images/Images.js [hunk 9]
```diff
@@ -40,125 +52,105 @@ export const CloudinaryImage = ({
...
-      ) {
-        let imageSettings = {
-          ...baseImageSettings,
-          format: fileFormats[format],
-          dpr: '1.0',
-        }
+        return cld.url(filePath(), imageSettings) + ` ${indx + 1}x`
+      })
+
+      let minMax = breakpoint < mediaBreakpoints.length - 1 ? `min` : `max`
```
Comment (roblevintennis) [inline src/components/Images/Images.js]: Prefer `const`

File: src/components/Images/Images.md [hunk 10]
```diff
@@ -8,23 +8,23 @@
...
+  <div style={{ height: 100 }}>Test lazy loading</div>
   <CloudinaryImage
     crop="fill"
     alt="father and kids playing"
     width={[250,350,450,550]}
     height={[250,350,450,550]}
     publicId="https://res.cloudinary.com/getethos/image/upload/v1565712179/01_NEW%20Lifestyle%20%28Rebrand%29/life-insurance-father-and-kids-playing.jpg"
   />
-  <div style={{ height: 1000 }}>Test lazy loading</div>
+  <div style={{ height: 100 }}>Test lazy loading</div>
```
Comment (roblevintennis) [inline src/components/Images/Images.md]: Remind me what unit this maps to? % or vh or something?

**Comment by roblevintennis** [comment]
@tkarssli I generally like the changes you've made so far! Left some nitpicky comments you can decide if you want to implement or not. It's still showing as [WIP] so I'll hold off on a final review, but thanks for taking this on! 🙌

File: src/components/Images/Images.js (hunk 8 above)
Comment (tkarssli) [reply src/components/Images/Images.js]: constantized

File: src/components/Images/Images.js (hunk 9 above)
Comment (tkarssli) [reply src/components/Images/Images.js]: constantized

**Comment by tkarssli** [comment]
Not sure why the travis is failing

File: src/components/Images/Images.js [hunk 11]
```diff
@@ -1,30 +1,150 @@
...
+        media={`(${minMax}-width: ${mediaBreakpoints[breakpoint]}px)`}
+        data-srcset= {srcsetData.join(', ')}
+        type={`image/${format}`}
+      />)
+      
+    }
+    return sourceTags
+  }
+
+  const isSvg = (publicId) => {
```
Comment (roblevintennis) [inline src/components/Images/Images.js]: Non Blocking: Do we think SVG specific stuff should be pushed into it's own custom hook or module? If so, I think we should do it as a follow up task and not block PR

File: src/components/Images/Images.js [hunk 12]
```diff
@@ -51,13 +171,24 @@ CloudinaryImage.CROP_METHODS = {
   CROP: 'crop',
 }
 
-CloudinaryImage.propTypes = {
-  // We're trying to avoid/reduce className restyling, but let's try it here:
+CloudinaryImage.IMAGE_FILE_TYPES = {
```
Comment (roblevintennis) [inline src/components/Images/Images.js]: Non Blocking:
My sense is we'll eventually do the same separating of concerns for image types e.g. invert the control. But I don't have a clear recommendation for how to do that yet.

File: src/components/Images/Images.md [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+  ZipInput,
+  EmailInput,
+} from '../index'
+
+;<div style={{ width: 400 }}>
+  <TitleSmall.Serif.Book500>Minimal Props</TitleSmall.Serif.Book500>
+  <Spacer.H24/>
+  <CloudinaryImage
+    alt="father and kids playing"
+    publicId="https://res.cloudinary.com/getethos/image/upload/v1565712179/01_NEW%20Lifestyle%20%28Rebrand%29/life-insurance-father-and-kids-playing.jpg"
```
Comment (roblevintennis) [inline src/components/Images/Images.md]: How do we know these images won't go away?

File: src/components/Images/Images.scss [hunk 14]
```diff
@@ -3,14 +3,14 @@
...
 }
 
 .Image {
   display: block;
-
-  // These are consequential, may need to pull out.
+  max-width: 100%;
   width: 100%;
-  height: auto;
+  object-fit: cover;
```
Comment (roblevintennis) [inline src/components/Images/Images.scss]: Non Blocking:

Hrm, do we need to support other `object-fit` properties? I was looking at the MDN for what these was and I could see us wanting some of the others.

File: src/components/Images/__snapshots__/Images.spec.js.snap [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+    media="(min-width: 600px)"
+    type="image/webp"
+  />
+  <source
+    data-srcset="https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_1.0,fl_progressive:semi,h_100,q_auto:eco,w_100/v1/something.com/otherthing.webp 1x, https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_2.0,fl_progressive:semi,h_100,q_auto:eco,w_100/v1/something.com/otherthing.webp 2x, https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_3.0,fl_progressive:semi,h_100,q_auto:eco,w_100/v1/something.com/otherthing.webp 3x"
+    media="(max-width: 599px)"
+    type="image/webp"
+  />
+  <source
+    data-srcset="https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_1.0,fl_progressive:semi,h_400,q_auto:eco,w_400/v1/something.com/otherthing.jp2 1x, https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_2.0,fl_progressive:semi,h_400,q_auto:eco,w_400/v1/something.com/otherthing.jp2 2x, https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_3.0,fl_progressive:semi,h_400,q_auto:eco,w_400/v1/something.com/otherthing.jp2 3x"
```
Comment (roblevintennis) [inline src/components/Images/__snapshots__/Images.spec.js.snap]: Same concern on if we'll need theseee

File: src/components/design-system.css [hunk 16]
```diff
@@ -161,14 +161,16 @@ textarea {
...
 figure {
   margin: 0;
 }
 
 .Image {
   display: block;
+  max-width: 100%;
   width: 100%;
-  height: auto;
+  object-fit: cover;
```
Comment (roblevintennis) [inline src/components/design-system.css]: It would be nice if this stuff could be true CSS Modules

File: src/components/Images/Images.scss [hunk 17]
```diff
@@ -3,14 +3,14 @@
 
 // WIP.
 
+/* Generally used as a wrapper around Image components. */
 figure {
   margin: 0;
 }
 
 .Image {
```
Comment (roblevintennis) [inline src/components/Images/Images.scss]: It would be nice if this stuff could be true CSS Modules and not a global

File: src/components/Images/Images.scss (hunk 14 above)
Comment (tkarssli) [reply src/components/Images/Images.scss]: I think it should be overridable if the user wants to pass through a classname and apply custom css. The reason I had to add this was to make it backwards compatible with the previous image component. Without it almost all the images on the CSS were getting horizontally skewed on smaller screens.

File: src/components/Images/Images.md (hunk 13 above)
Comment (tkarssli) [reply src/components/Images/Images.md]: Yeah these could be removed anytime, what would be an appropriate image to use?

File: src/components/design-system.css (hunk 16 above)
Comment (tkarssli) [reply src/components/design-system.css]: I will try converting this to css modules

Comment (roblevintennis) [reply src/components/design-system.css]: Ok, I'm fine if you'd like to do on a follow up though. Up to you

**Comment by tkarssli** [comment]
Converted over to CSS modules.

File: src/components/Images/Images.md (hunk 13 above)
Comment (kenning) [reply src/components/Images/Images.md]: because our image elements are baking in cloudinary dependence i don't see any other way to do this.

**Comment by jameshawkinsjr** [comment]
Need to fix banner module refactor to not allow images(?) vs. svg

Screen Shot 2019-11-01 at 4 48 12 PM

File: src/components/Images/Images.js [hunk 18]
```diff
@@ -1,30 +1,151 @@
 import React from 'react'
 import PropTypes from 'prop-types'
-import * as Cloudinary from 'cloudinary-react'
+import cloudinary from 'cloudinary-core'
```
Comment (anbright) [inline src/components/Images/Images.js]: nit: re-arrange imports

File: src/components/Images/Images.js [hunk 19]
```diff
@@ -1,30 +1,151 @@
...
+  // Verify that no invalid props were supplied
+  const [includesInvalid] = useInvalid(
+    Object.keys(CloudinaryImage.PUBLIC_PROPS)
+  )
+  includesInvalid(rest)
+
+  const { WEBP, JP2, JPEG, SVG } = CloudinaryImage.IMAGE_FILE_TYPES
+  const baseImageSettings = {
+    quality: 'auto:eco',
+    crop: crop,
```
Comment (anbright) [inline src/components/Images/Images.js]: nit: object property shorthand

File: src/components/Images/Images.js [hunk 20]
```diff
@@ -1,30 +1,151 @@
...
+    secure: true,
+    flags: ['progressive:semi'],
+  }
+  let imageClasses = ['lazyload', className]
+  let reverseWidth, reverseHeight
+  width && (reverseWidth = width.slice().reverse())
```
Comment (anbright) [inline src/components/Images/Images.js]: This code is setting `reverseWidth` to `width.slice().reverse()` if `width` is defined, correct? It is a little confusing so using `if` blocks may be better

File: src/components/Images/__snapshots__/Images.spec.js.snap [hunk 21]
```diff
@@ -0,0 +1,71 @@
...
+    media="(max-width: 599px)"
+    type="image/jpeg"
+  />
+  <img
+    alt="alt text"
+    className="Image lazyload testImage"
+    data-src="https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_1.0,fl_progressive:semi,q_auto:eco/v1/something.com/otherthing.jpeg"
+  />
+</picture>
+`;
```
Comment (kenning) [inline src/components/Images/__snapshots__/Images.spec.js.snap]: really sorry to pile on here, but i have a different concern with this pr. right off the bat, it seems very strange to me that the optimal way to send images would be 12 source tags, each with a data-src attribute over 400 characters long, and an img tag. 

this single image is over 6400 characters, or about 6.4kb, of just html. the front page has 22  tags, each of which would be replaced with 6.4kb of html. 22 * 6.4kb = 140.8kb. our current front page is 189kb (before gzip), so this pr would almost double the size of the html being transferred. given that one of our OKR's is first contentful paint i would really like to see some timing on this.

if you look at the [cloudinary docs](https://cloudinary.com/documentation/image_transformations#automatic_format_selection) they have "format=auto" and "quality=auto" settings which offload most of the work that this file is doing. they also pepper in odd little 'tips' like this one:

> “When automatic format (f_auto) is used together with automatic quality (q_auto), the PNG format may be selected when the automatic quality algorithm decides that it better fits the specific image. This allows delivering better looking and economical image files.”

so first of all, this goes against what i thought i knew about image formats -- i was unaware that png's would ever be faster than jpeg's, let alone these weird new formats like webp/jpeg-2000. and right now our generated html never makes a png, so either they are wrong or there are times when we're fetching something less than optimal. 

this goes back to an early thought I had about this pr, which is that it feels like we're duplicating the work that cloudinary is supposed to be doing for us of choosing the best format and quality settings based on device size, os, etc. without performance testing we have no way of knowing if we are right or not; and maybe this is laziness but I sort of assume that cloudinary will generally do a better job than we will, because this is one of the main selling points for their business.

**tl;dr** before we merge this I think we should do some quick tests to see if this is indeed faster than letting cloudinary determine format and quality on a handful of browsers. the important things to measure would be first contentful paint (due to our okr) and the final above-the-fold page load time.

File: src/components/Images/Images.js (hunk 11 above)
Comment (zach-ethos) [reply src/components/Images/Images.js]: Yeah I would vote for not making this support SVGs unless there is a specific reason. The motivator for Cloudinary is the difficulty of handling/serving large raster files basically.

File: src/components/Images/__snapshots__/Images.spec.js.snap (hunk 21 above)
Comment (zach-ethos) [reply src/components/Images/__snapshots__/Images.spec.js.snap]: Agree^

File: src/components/Images/Images.js (hunk 20 above)
Comment (roblevintennis) [reply src/components/Images/Images.js]: I agree here. I'd say unless we're in JSX we should shoot for readability first; especially since we have an assignment happening

File: src/components/Images/Images.module.scss [hunk 22]
```diff
@@ -20,3 +21,14 @@ figure {
   width: 100%;
   height: auto;
 }
+
+.blur-up {
+  -webkit-filter: blur(5px);
+  filter: blur(5px);
```
Comment (roblevintennis) [inline src/components/Images/Images.module.scss]: Isn't autoprefixer gonna do this for us?

image

File: src/components/Images/__snapshots__/Images.spec.js.snap (hunk 21 above)
Comment (tkarssli) [reply src/components/Images/__snapshots__/Images.spec.js.snap]: @kenning @zach-ethos Okay I changed up this PR a little bit. It should now use the automatic format setting that cloudinary offers and through testing it looks like it works pretty well, this was a good call. I had to leave in the DPR setting as the auto setting in their lib was not working as well manual. 
I also used this opportunity to add in LQIP to the images which should help with first loads and content shifting around. I added a blur-up to make the transition a little smoother between the images.
This is the gist of what we build now:
* 4 source tags, one for each media query.
* 1 img tag that has the LQIP as `srcSet` and also a fallback image for old browsers in the `src`

```html
<picture>
  <source media="(min-width: 1200px)" data-srcset="https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_1.0,f_auto,fl_progressive:semi,h_624,q_auto:eco,w_636/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg 1x, https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_2.0,f_auto,fl_progressive:semi,h_624,q_auto:eco,w_636/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg 2x, https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_3.0,f_auto,fl_progressive:semi,h_624,q_auto:eco,w_636/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg 3x"/>
  <source media="(min-width: 900px)" data-srcset="https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_1.0,f_auto,fl_progressive:semi,h_428,q_auto:eco,w_428/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg 1x, https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_2.0,f_auto,fl_progressive:semi,h_428,q_auto:eco,w_428/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg 2x, https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_3.0,f_auto,fl_progressive:semi,h_428,q_auto:eco,w_428/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg 3x"/>
  <source media="(min-width: 600px)" data-srcset="https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_1.0,f_auto,fl_progressive:semi,h_332,q_auto:eco,w_332/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg 1x, https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_2.0,f_auto,fl_progressive:semi,h_332,q_auto:eco,w_332/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg 2x, https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_3.0,f_auto,fl_progressive:semi,h_332,q_auto:eco,w_332/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg 3x"/>
  <source media="(max-width: 599px)" data-srcset="https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_1.0,f_auto,fl_progressive:semi,h_250,q_auto:eco,w_375/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg 1x, https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_2.0,f_auto,fl_progressive:semi,h_250,q_auto:eco,w_375/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg 2x, https://res.cloudinary.com/getethos/image/upload/c_fill,dpr_3.0,f_auto,fl_progressive:semi,h_250,q_auto:eco,w_375/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg 3x"/>
  <img class="Images-module--Image--1EVuH  lazyload " src="https://res.cloudinary.com/getethos/image/upload/c_fill,f_auto,fl_progressive:semi,q_auto:eco,t_unsupported/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg" srcSet="https://res.cloudinary.com/getethos/image/upload/c_fill,f_auto,fl_progressive:semi,h_250,q_auto:eco,t_lqip,w_375/v1560273411/01_Lifestyle/life-insurance-01-family-playing-in-bed.jpg" alt="Family playing in bed"/>
</picture>
```

As you can see this is ~2900 characters per image, vs the ~9k from before with 12 source tags. The impact is very low from my testing on FCP. A page with many images like `/blog` comes out to about 50k more characters which is zipped to ~700 bytes. 

I went pretty soft on LQIP so they are usually between 5 -10 kb each. But we could take a more aggressive approach using either a simple default placeholder, or using greyscale versions of the images . (~1.5 - 3 kb).

File: src/components/Images/Images.module.scss (hunk 22 above)
Comment (tkarssli) [reply src/components/Images/Images.module.scss]: Fixed

File: src/components/Images/Images.js (hunk 20 above)
Comment (tkarssli) [reply src/components/Images/Images.js]: I needed to dupe width then set it to be the reversed version. Ill move it into an if statement

File: src/components/Images/Images.js (hunk 19 above)
Comment (tkarssli) [reply src/components/Images/Images.js]: fixed

File: src/components/Images/__snapshots__/Images.spec.js.snap (hunk 21 above)
Comment (kenning) [reply src/components/Images/__snapshots__/Images.spec.js.snap]: we medium now

File: src/components/Images/Images.module.scss [hunk 23]
```diff
@@ -0,0 +1,32 @@
+/* @getethos/design-system/Images.scss
+   ========================================================================== */
+
+// WIP.
```
Comment (kenning) [inline src/components/Images/Images.module.scss]: My test caught something! 😃

Monorepo webpack blows up if you use `//` syntax for comments.

Comment (kenning) [reply src/components/Images/Images.module.scss]: i'm pushing some fixes rn

**Comment by kenning** [comment]
Here's a video of the blurring in action: https://i.gyazo.com/292d79d9f299504997e2daac342ef944.mp4

It looks pretty good to me. They don't take up space until they're done loading, i couldn't tell if this was expected behavior or not but this is how other webpages work so i think its ok...

**Comment by tkarssli** [comment]
> Here's a video of the blurring in action: https://i.gyazo.com/292d79d9f299504997e2daac342ef944.mp4
> 
> It looks pretty good to me. They don't take up space until they're done loading, i couldn't tell if this was expected behavior or not but this is how other webpages work so i think its ok...

Basically there is a non lazyloaded placeholder for every picture on a page, so there is going to be a little bit of a pop in there. But ideally this placeholder image is verrrry small so it should be brief. Then the browser can continue loading the full res image and replace it.

File: src/components/design-system.css (hunk 16 above)
Comment (tkarssli) [reply src/components/design-system.css]: converted

File: src/components/Images/Images.scss (hunk 17 above)
Comment (tkarssli) [reply src/components/Images/Images.scss]: converted

File: src/components/Images/Images.js (hunk 18 above)
Comment (tkarssli) [reply src/components/Images/Images.js]: rearranged

**Comment by zach-ethos** [comment]
> As you can see this is ~2900 characters per image, vs the ~9k from before with 12 source tags. The impact is very low from my testing on FCP. A page with many images like /blog comes out to about 50k more characters which is zipped to ~700 bytes.

@tkarssli Not sure I get it – like, in this PR, one typical CloudinaryImage would build to ~3kb? And before it was ~9kb? Or was it ~9x12kb.

Either way if this increases the page weight we probably should quantify the benefit we're getting more. E.g. are the images smaller? Blur-up is a big qualitative benefit. I could buy that gzip makes this not a big deal though.

**Comment by tkarssli** [comment]
> @tkarssli Not sure I get it – like, in this PR, one typical CloudinaryImage would build to ~3kb? And before it was ~9kb? Or was it ~9x12kb.

A single image would have piture/source tags that amounted to 9kb unzipped.

>Either way if this increases the page weight we probably should quantify the benefit we're getting more. E.g. are the images smaller? Blur-up is a big qualitative benefit. I could buy that gzip makes this not a big deal though.

From testing the weight added is negligible when compared to how much less data is being transferred as a result of the lazy loading

**Comment by kenning** [comment]
bias for action!
